### PR TITLE
Abstract Text Editor: don't change StyledText by CSS.

### PR DESCRIPTION
**Pull request by DenisUngemach**
Fix for #869

Change-Id: Ib739afdc301c4e37dcbbe37ef063ed32cb7f53d7

### CODE REVIEW

**Comment by DenisUngemach** [comment]
In this change i just deactivate the CSS stylesheet, which overrides the colors from the "Text Editors" preferences.
Can someone confirm whether this is the right way to solve #869 .

And by the way issue #869 also affects the java class editor. That can be checked with the background color.

**Comment by BeckerWdf** [comment]
> Fix for #869
> 
> Change-Id: Ib739afdc301c4e37dcbbe37ef063ed32cb7f53d7

We don't use Gerrit. So we don't need a Change-Id

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java [hunk 1]
```diff
@@ -2277,6 +2277,13 @@ private int computeOffsetAtLocation(ITextViewer textViewer, int x, int y) {
 	/** The width of the vertical ruler. */
 	protected static final int VERTICAL_RULER_WIDTH= 12;
 
+	/*
+	 * The colors of AbstractTextEditor can be configured by the "Text Editors"
+	 * preferences. This color configuration will be changed by theming. This means
+	 * no need for CSS basecolors here.
+	 */
+	private static final String DISABLE_CSS = "org.eclipse.e4.ui.css.disabled"; //$NON-NLS-1$
```
Comment (BeckerWdf) [inline bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java:2279]: here does this "magic" constant come from?

Comment (DenisUngemach) [reply bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java:2279]: From org.eclipse.e4.ui.css.swt: AbstractCSSSWTEngineImpl line 117

**Comment by DenisUngemach** [comment]
This feature has already been used in: 
See EGit's [UIUtils](https://git.eclipse.org/r/c/egit/egit/+/190575/2/org.eclipse.egit.ui/src/org/eclipse/egit/ui/UIUtils.java) class

**Comment by vogella** [comment]
@BeckerWdf what is the status here with the review? Did you test the change?

**Comment by BeckerWdf** [comment]
@DenisUngemach will update his PR. I then will do a manual test.

**Comment by sratz** [comment]
In princple, this works well.

But we were wondering whether disabling CSS completely is the way to go. The preferences only deal with foreground + background colors, a CSS might do more than just that.

In theory that is - in practice, the CSS found in Platform only deals with foreground + background colors as well.

So we are probably safe here and if there are no objections we can start with this and wait for feedback.

If needed, we could maybe later introduce a more fine-granular E4 property, such as:

```
setData("org.eclipse.e4.ui.css.disabledProperties", "background-color,foreground-color")
```

However, your alternative PR #1349 also has an additional condition `getPreferenceStore() != null`
https://github.com/eclipse-platform/eclipse.platform.ui/pull/1349/files#diff-c37eb3db119ede5b1695bb0217864648bfa40dfaafcfa25c7bbd7131d69ed641R3401

That might be useful to add as a safeguard in case `AbstractTextEditor` is used in a non-platform context without attached preference (is that even possible?). In that case, we wouldn't want to disable CSS.

**Comment by vogella** [comment]
I don't think the extra check is necessary.

> a CSS might do more than just that.

As for the colors of the syntax highligthing, we manipulate the preference values directly, so we should be ok.

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java [hunk 2]
```diff
@@ -3366,6 +3373,8 @@ public void createPartControl(Composite parent) {
 		int styles= SWT.V_SCROLL | SWT.H_SCROLL | SWT.MULTI | SWT.BORDER | SWT.FULL_SELECTION;
 		fSourceViewer= createSourceViewer(parent, fVerticalRuler, styles);
 
+		fSourceViewer.getTextWidget().setData(DISABLE_CSS, Boolean.TRUE);
```
Comment (BeckerWdf) [inline bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java:3374]: In @vogella 's proposal there was a nice comment why it's ok to disable the styling. Can you add a similar comment?

**Review by BeckerWdf — CHANGES_REQUESTED** [review]

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java (hunk 2 above)
Comment (DenisUngemach) [reply bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java:3374]: Done

**Comment by BeckerWdf** [comment]
I did a manual test on macOS.
Without this change I can reproduce this issue 
With this change the issue is gone. Change to background and foreground color in the TextEditors preference page are not correctly handled. I teste editor close and re-opening as well as workbench restart.
All looks good for me.

**Comment by vogella** [comment]
GH says, you still require changes @BeckerWdf ?

**Comment by BeckerWdf** [comment]
> GH says, you still require changes @BeckerWdf ?

Let's wait for the build..

**Comment by vogella** [comment]
> > GH says, you still require changes @BeckerWdf ?
> 
> Let's wait for the build..

We (should) always wait for the build. I wanted to know if you still require changes in this PR? @DenisUngemach could finish it  in this case.

**Review by BeckerWdf — APPROVED** [review]

**Comment by BeckerWdf** [comment]
Are version number increases necessary for this change?

**Comment by vogella** [comment]
> Are version number increases necessary for this change?

No, the build would fails if version increases are necesarry. I did not check if the test errors relate to this change.

**Comment by BeckerWdf** [comment]
> No, the build would fails if version increases are necessary. 

So has the version number already been increased?

> I did not check if the test errors relate to this change.

The seem to be unrelated to me.

**Comment by BeckerWdf** [comment]
Test failures are not related and already reported in https://github.com/eclipse-platform/eclipse.platform.ui/issues/1351

**Comment by vogella** [comment]
Cool, thanks @DenisUngemach and @BeckerWdf

**Comment by DenisUngemach** [comment]
> Cool, thanks @DenisUngemach and @BeckerWdf

You're welcome. Actually my first idea was to create a custom css configuration: "[CSS support for custom widgets](https://www.vogella.com/tutorials/Eclipse4CSS/article.html#css-support-for-custom-widgets)"for the AbstractTextEditor and set there the color configuration. But this didn't work that well because somewhere in the compatibility handler css class and css id will be defined:

image

And i only wanted to write the css class. Is this css ID writing justified? Maybe these base layers should override css class, but i think the css ID is over the top.

Best regards,
Denis

**Pull request merged**